testcode/lib/clock-test.c

```diff
@@ -0,0 +1,100 @@
+/* Clock test program
+ *
+ * 25-Sep-2018, chris@groessler.org
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <errno.h>
+
+#ifdef __CC65__
+#include <conio.h>
+#include <cc65.h>
+
+static void exitfn(void)
+{
+    if (doesclrscrafterexit()) cgetc();
+}
+#endif /* #ifdef __CC65__ */
+
+static void print_time(void)
+{
+    struct tm *cur_tm;
+    time_t cur_time = time(NULL);
+    if (cur_time == -1) {
+        printf("time() failed: %s\n", strerror(errno));
+        return;
+    }
+    cur_tm = localtime(&cur_time);
+
+    printf("time: %s\n", asctime(cur_tm));
+    // DEBUG:
+    printf("mday=%d mon=%d year=%d\nhour=%d min=%d sec=%d\n", cur_tm->tm_mday, cur_tm->tm_mon, cur_tm->tm_year, cur_tm->tm_hour, cur_tm->tm_min, cur_tm->tm_sec);
+}
+
+int main(int argc, char **argv)
+{
+    static char c;
+    static int s;
+    static struct tm cur_time;
+    static struct timespec new_time;
```

Comment:
If I'm not mistaken these four `static` don't make sense here, do they?

Reply:
See https://cc65.github.io/doc/cc65.html#toc7.16
Here I wrote the "static" explicitly instead of using the pragma.

Reply:
But why do you want to have those variables in the BSS segment instead of the stack in the first place?

Reply:
Thanks for pointing out the cc65 docs to me ;-))

With that argument all cc65 programs should be built that way. I think that this type of "general optimizationl" doesn't belong to test/sample code. I'd prefer if you'd  remove it.

Reply:
Nitpicking, eh? What about these:


    $ find samples/ testcode/ -type f -name \*.c | xargs grep -n pragma | grep static
    samples/sieve.c:41:#pragma static-locals(1);
    samples/fire.c:59:#pragma static-locals (1);
    samples/mandelbrot.c:43:#pragma static-locals (1);
    samples/plasma.c:54:#pragma static-locals (1);
    $


Anyway, I've changed it...

Reply:
> Nitpicking?

I don't think so. What you referenced...
* does not change the source to deviate from what one expects from a normal C program - and therefore don't distract the user from understanding it. The exsistance of this thread proves that the 'static's in the source code do so. They make readers like me wonder why your program may depend on those variables sitting in the BSS - maybe because of yet another banking mystery?
* does actually need the speed they get out static locals - while the program at hand doesn't.

Reply:
Fair enough.

Reply:
Thanks for your understanding - and the change :-)